[dynamo][user-defined] Fix bugs with method descriptors

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #139914
* __->__ #139856

Should fix some problems in https://github.com/pytorch/pytorch/pull/138080

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

anijain2305 committed:
[dynamo][user-defined] Fix bugs with method descriptors

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][user-defined] Fix bugs with method descriptors"


Should fix some problems in https://github.com/pytorch/pytorch/pull/138080

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

mlazos commented on torch/_dynamo/variables/user_defined.py:
```diff
@@ -1107,20 +1107,25 @@ def var_getattr(self, tx: "InstructionTranslator", name):
+                    AttrSource(self.cls_source, "__dict__"), name
+                )
+                descriptor_get_source = AttrSource(descriptor_source, "__get__")
+            descriptor_var = UserDefinedObjectVariable(subobj, source=descriptor_source)
```
Can we use builder here?

anijain2305 replied:
For sourceless, I cant rely on the SourcelessBuilder because it is a UserDefinedObject. But, let me try it for sourceful.

anijain2305 replied:
Done

anijain2305 committed:
Update on "[dynamo][user-defined] Fix bugs with method descriptors"


Should fix some problems in https://github.com/pytorch/pytorch/pull/138080

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]
